src/accordion/section.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import uniqueId from '../utils/unique-id.js';
+import styles from './section.css';
+
+const headings = {
+  1: 'h1',
+  2: 'h2',
+  3: 'h3',
+  4: 'h4',
+  5: 'h5',
+  6: 'h6',
+};
+
+export default class Section extends React.Component {
+  constructor() {
+    super();
+    this.accessibleId = uniqueId();
+  }
+
+  render() {
+    const {
+      cannotClose,
+      children,
+      cssContainer,
+      cssHeading,
+      cssPanel,
+      cssTrigger,
+      headingLevel,
+      onClick,
+      open,
+      title,
+    } = this.props;
+    const titleVal = typeof title === 'function' ? title({ open }) : title;
+    const Heading = headings[headingLevel];
+
+    if (!Heading) {
+      throw new Error('Accordion must have a headingLevel between 1 and 6');
+    }
+
+    return (
+      <div className={cssContainer}>
+        <Heading className={cssHeading}>
+          <button
+            aria-controls={this.accessibleId}
+            aria-disabled={cannotClose}
+            aria-expanded={open}
+            className={cssTrigger}
+            type="button"
+            onClick={onClick}
+          >
+            {titleVal}
+          </button>
+        </Heading>
```

Comment:
Note the structure here of:

```jsx
<Heading>
  <Button />
</Heading>
```

Part of me wants to make this whole sequence a render prop and let users set whatever DOM they want. 

However, [the authoring practices](https://www.w3.org/TR/wai-aria-practices-1.1/#accordion) are clear that the only child of the "heading" should be the "button", and it made more sense to me to be strict.